controllers/weather.js

```diff
@@ -11,12 +13,71 @@ module.exports.getWeatherIndex = async (req, res, next) =>
 }
 
 
-// @desc     Get weather data
-// @router  GET /weather/:city
-// @access  public
+/**
+ * Get city weather
+ * @param {*} req 
+ * @param {*} res 
+ * @param {*} next 
+ */
 module.exports.getCityWeather = async (req, res, next) =>
 {
-    const city = req.params.city;
-    const result = await fetchWeatherDataByCity(city);
-    res.render('weather/byCity', { title: `Weather in ${ city }`, data: result });
+    try
+    {
+        const city = req.params.city;
+        const weatherResult = await fetchWeatherDataByCity(city);
+
+        res.status(200).json({ success: true, data: weatherResult })
+    } catch (error)
+    {
+        next(error)
```

Comment:
why don't we respond error here but using `next(error)`

Reply:
it will be handle on middleware :)

Reply:
nice! 👍

Reply:
we might not need to do try/catch like, express might already catch it by default. 
https://expressjs.com/en/guide/error-handling.html

```
Starting with Express 5, route handlers and middleware that return a Promise will call next(value) automatically when they reject or throw an error. For example:...
```

Reply:
We still use express 4 though.